### PR TITLE
MockTracer use text map propag as default constructor

### LINKED ISSUE #177: MockTracer use TextMap propagator by default
`MockTracer`, currently uses no/"printer" propagator by default. A better default is to use text map propagator.

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #177

### CODE REVIEW

File: opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java [hunk 1]
```diff
@@ -39,16 +39,12 @@
...
+    private final List<MockSpan> finishedSpans = new ArrayList<>();
     private final Propagator propagator;
-    private ActiveSpanSource spanSource;
+    private final ActiveSpanSource spanSource;
 
     public MockTracer() {
-        this(new ThreadLocalActiveSpanSource(), Propagator.PRINTER);
-    }
-
-    public MockTracer(ActiveSpanSource spanSource) {
```
Comment (pavolloffay) [inline opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java]: removing this to keep the API clean.

**Comment by pavolloffay** [comment]
If there are no issues I will merge in 48h.

cc @opentracing/otsc @opentracing/otiab

**Comment by pavolloffay** [comment]
anybody from @opentracing/otsc to approve the PR?

**Comment by carlosalberto** [comment]
@pavolloffay Was thinking if we should re-target this to 0.31 directly (as that's the new defacto 'upstream' branch). Let me know, as I could re-target it myself (so we can enjoy it for RC 2 ;) )

**Comment by carlosalberto** [comment]
As a reference, I re-targeted this patch for v0.31.0 - and the result is here: https://github.com/pavolloffay/opentracing-java/commits/master. @pavolloffay hope this helps to get it integrated soon ;)

**Comment by pavolloffay** [comment]
> Was thinking if we should re-target this to 0.31 directly (as that's the new defacto 'upstream' branch). Let me know, as I could re-target it myself (so we can enjoy it for RC 2 ;) )

@carlosalberto what about rebasing `v0.31` against the master before publishing RC2?

**Comment by carlosalberto** [comment]
@pavolloffay That's actually a good idea. So, lets merge this change and to the rebase later today.

**Comment by pavolloffay** [comment]
+1 IIR there are more commits in master than in `v0.31.`. I will merge

**Pull request merged**